### PR TITLE
Exchange deprecated systemd resources with the appropriate for cgroupv2

### LINKED ISSUE #2164: cgroup2: systemd manager uses deprecated options
https://github.com/opencontainers/runc/blob/e57a774066af2f0adc76ffb6201588091cd872d5/libcontainer/cgroups/systemd/unified_hierarchy.go#L99-L132

According to [`systemd.resource-control(5)`](https://www.freedesktop.org/software/systemd/man/systemd.resource-control.html), `CPUShares`, `MemoryLimit`, and `BlockIOWeight` are deprecated

### PR DESCRIPTION
**Pull request by Zyqsempai**
Fix #2164 

Exchanged deprecated systemd resources with the appropriate for cgroupv2

Signed-off-by: Boris Popovschi <zyqsempai@mail.ru>

### CODE REVIEW

File: libcontainer/cgroups/systemd/unified_hierarchy.go [hunk 1]
```diff
@@ -117,11 +117,6 @@ func (m *UnifiedManager) Apply(pid int) error {
 			newProp("CPUQuotaPerSecUSec", cpuQuotaPerSecUSec))
 	}
 
-	if c.Resources.BlkioWeight != 0 {
-		properties = append(properties,
-			newProp("BlockIOWeight", uint64(c.Resources.BlkioWeight)))
```
Comment (AkihiroSuda) [inline libcontainer/cgroups/systemd/unified_hierarchy.go]: Do we have modern equivalent for this?

Comment (Zyqsempai) [reply libcontainer/cgroups/systemd/unified_hierarchy.go]: Yes, it's `IOWeight` but I tested it and faced some strange error `no such device or address`, so I will continue investigation related to this error, but until that we can merge this part.

Comment (AkihiroSuda) [reply libcontainer/cgroups/systemd/unified_hierarchy.go]: If systemd can translate the legacy value, let's keep the legacy code for IO.

**Comment by mrunalp** [comment]
LGTM

**Comment by AkihiroSuda** [comment]
@filbranden @giuseppe PTAL?

**Comment by Zyqsempai** [comment]
@filbranden @giuseppe Ping

**Comment by giuseppe** [comment]
LGTM but I am not a maintainer

**Comment by filbranden** [comment]
Hey!

I haven't had time to look deeply into it. A problem in the past was whether older versions of systemd would have problems with this change.

I guess if you can test this on something like RHEL 7 / CentOS 7 (systemd v219 + some backports) to ensure this won't break on a systemd that old, that would be great!

You mentioned cgroupv2, so CentOS 7 doesn't really support cgroupv2, so it's not relevant there. (I haven't looked deeply into the code changes to check whether this affects cgroupv1 or cgroupv2 only.)

Having said that, otherwise, LGTM. As long as those issues are not of concern, you should be fine! Thanks for this!

Cheers,
Filipe

**Comment by Zyqsempai** [comment]
@hqhq @crosbymichael PTAL and let's merge it;)

**Comment by Zyqsempai** [comment]
@cyphar PTAL here also, it should be merged long time ago;)

**Comment by Zyqsempai** [comment]
@mrunalp @dqminh maybe you can take a look at this one?

**Comment by mrunalp** [comment]
LGTM

**Comment by hqhq** [comment]
LGTM

**Pull request merged**